### DISCUSSION TITLE
Accessing android system in beeware app

### DISCUSSION THREAD

**Discussion by akgalp** [beeware/beeware#234] [Q&A]
Hello,
Newbie android dev here.
I'm trying to use android system in beeware app.  (sensors, camera, wifi, usb and bluetooth)
Tried using 'plyer' with 'pyjnius' but it seems chaquopy does not support pyjnius. 
Is there an example to follow and what packages need to be installed?

Any help appreciated and thanks in advance.

**Comment by freakboy3742** [comment]
At present, we don't have a cross-platform layer for accessing device hardware; this is on our todo list.

You can use the platform native APIs to access system hardware (i.e., accessing the Android Java/Kotlin APIs directly), but that obviously isn't cross platform, and you'll need to become familiar with how you translate Java API documentation into the equivalent Python calls. The [Chaquopy](https://chaquo.com/chaquopy/doc/current/python.html) documentation can guide you on how this can be done.
